File: isma-intg-api/src/main/java/ru/nstu/isma/intg/api/methods/IntgMethod.java
package ru.nstu.isma.intg.api.methods;


import java.io.Serializable;

/**
 * @author Mariya Nasyrova
 * @since 30.08.14
 */
public interface IntgMethod extends Serializable {

    String getName();

    double nextY(double step, double[] k, double y, double f);

    StageCalculator[] getStageCalculators();

    AccuracyIntgController getAccuracyController();

    StabilityIntgController getStabilityController();

}

File: isma-intg-api/src/main/kotlin/ru/nstu/isma/intg/api/methods/IntgMethod.kt
package ru.nstu.isma.intg.api.methods

interface IntgMethod {
    val stageCalculators: Array<StageCalculator>?

    val accuracyController: AccuracyIntgController?

    val stabilityController: StabilityIntgController?

    fun nextY(step: Double, k: DoubleArray, y: Double, f: Double): Double
}

class IntegrationMethodRungeKutta(
    val stageCalculators: Array<StageCalculator> = emptyArray(),
    val accuracyController: AccuracyIntgController? = null,
    val stabilityController: StabilityIntgController? = null,
    val nextY: (step: Double, k: DoubleArray, y: Double, f: Double) -> Double
)